### PR TITLE
Add documentation for using code coverage

### PR DESCRIPTION
**Pull request by TLATER**
Hi, sorry for the out-of-the-blue PR, but we've been working on getting code coverage to run in remote execution downstream, and written up these docs.

The basic local workflow is fairly easy (though I think it's still useful to document how `genhtml` is intended to be used), but there are a handful of issues with remote execution. The doc as-is just lists and explains them, as well as workarounds, but we intend to fix at least some those problems over the coming weeks.

The python workflow is also quite rough at the moment, we're looking at making this simpler (see nedbat/coveragepy#587).

Nonetheless, it would be nice to have some review on this in the mean time, so that we can get it merged a bit more quickly, and at least have code coverage documented a bit better even if it is broken in some ways.

I think I may not have been diligent enough with checking whether it builds, so apologies if this just ends up causing a build failure. I hope CI will help a little with this.

### CODE REVIEW

**Comment by TLATER** [comment]
@lfpino has had a glance over this before, can't seem to request a review from him, so I'll just add him here.

File: site/docs/coverage.md
Comment (lfpino) [inline site/docs/coverage.md:132]: nit: could you file issues for these TODOs once you get the approval from the Bazel team?

**Review by lfpino — APPROVED** [review]

File: site/docs/coverage.md
Comment (TLATER) [reply site/docs/coverage.md:132]: Yes, of course :)

File: site/docs/coverage.md
Comment (c-mita) [inline site/docs/coverage.md:115]: Might be worth leaving a note regarding C++ as well (which should also "just work", assuming the required tools are available).

**Review by c-mita — COMMENTED** [review]
This looks fine; I just have one comment re. C++, but I don't think it's pressing if you don't feel you can answer it.

File: site/docs/coverage.md
Comment (TLATER) [reply site/docs/coverage.md:115]: We planned on testing/documenting that too, originally. @lfpino has used it for another project, but it's in a similar state as Python as far as we can tell, with environment variables being thrown around for the tooling. We wanted to make sure we actually have a minimal working example before doing so, though, and it sort of fell under the radar.

It's probably not a lot of work to try and replicate some coverage with C++, but I think I'd rather do that as a follow up - we're kind of at capacity, given holidays and such.

Comment (c-mita) [reply site/docs/coverage.md:115]: SGTM

**Review by c-mita — APPROVED** [review]

**Pull request closed without merging**
